File: services/web-api/src/main/java/com/github/jeffreyjpz/timestamped_for_f1_web_api/web/v1/meetings/dtos/Meeting.java
package com.github.jeffreyjpz.timestamped_for_f1_web_api.web.v1.meetings.dtos;

import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public class Meeting {

    @JsonAlias({"circuitKey", "circuit_key"})
    @JsonProperty("circuit_key")
    private final Integer circuitKey;

    @JsonAlias({"circuitShortName", "circuit_short_name"})
    @JsonProperty("circuit_short_name")
    private final String circuitShortName;

    @JsonAlias({"countryCode", "country_code"})
    @JsonProperty("country_code")
    private final String countryCode;

    @JsonAlias({"countryKey", "country_key"})
    @JsonProperty("country_key")
    private final Integer countryKey;

    @JsonAlias({"countryName", "country_name"})
    @JsonProperty("country_name")
    private final String countryName;

    @JsonAlias({"dateStart", "date_start"})
    @JsonProperty("date_start")
    private final String dateStart;

    @JsonAlias({"gmtOffset", "gmt_offset"})
    @JsonProperty("gmt_offset")
    private final String gmtOffset;

    @JsonAlias({"location"})
    @JsonProperty("location")
    private final String location;

    @JsonAlias({"meetingKey", "meeting_key"})
    @JsonProperty("meeting_key")
    private final Integer meetingKey;

    @JsonAlias({"meetingName", "meeting_name"})
    @JsonProperty("meeting_name")
    private final String meetingName;

    @JsonAlias({"meetingOfficialName", "meeting_official_name"})
    @JsonProperty("meeting_official_name")
    private final String meetingOfficialName;

    @JsonAlias({"year"})
    @JsonProperty("year")
    private final Integer year;

}

File: dashboard/src/types/meeting.ts
import z from "zod";

export const meeting = z
    .object({
        circuit_key: z.number().int().nonnegative(),
        circuit_short_name: z.string(),
        country_code: z.string().length(3),
        country_key: z.number().int().nonnegative(),
        country_name: z.string(),
        date_start: z.iso.datetime({ offset: true }),
        gmt_offset: z.string().regex(/^-?\d{2}:\d{2}:\d{2}$/),
        location: z.string(),
        meeting_key: z.number().int().nonnegative(),
        meeting_name: z.string(),
        meeting_official_name: z.string(),
        year: z.number().int().nonnegative(),
    });

export const meetings = z.array(meeting);

export const meetingRead = meeting
    .pick({
        meeting_key: true
    });

export const meetingsRead = meeting
    .extend({
        circuit_key: z.array(z.number().int().nonnegative()),
        circuit_short_name: z.array(z.string()),
        country_code: z.array(z.string().length(3)),
        country_key: z.array(z.number().int().nonnegative()),
        country_name: z.array(z.string()),
        date_start: z.array(z.iso.datetime({ offset: true })),
        gmt_offset: z.array(z.string().regex(/^-?\d{2}:\d{2}:\d{2}$/)),
        location: z.array(z.string()),
        meeting_key: z.array(z.number().int().nonnegative()),
        meeting_name: z.array(z.string()),
        meeting_official_name: z.array(z.string()),
        year: z.array(z.number().int().nonnegative()),
    })
    .partial();

export type Meeting = z.infer<typeof meeting>;
export type Meetings = z.infer<typeof meetings>;
export type MeetingRead = z.infer<typeof meetingRead>;
export type MeetingsRead = z.infer<typeof meetingsRead>;